CI failing on Ubuntu

https://github.com/spacetelescope/slitlessutils/pull/73 added a test that is passing on MacOS but is failing on Ubuntu. The 3.9 failure looks like version conflicts with Numpy, but the 3.11 failure appears to be a potentially real bug involving multiprocessing:

```
>       pdtfiles = tab(data, sources)  # noqa: F841

/home/runner/work/slitlessutils/slitlessutils/.tox/py311-test-cov/lib/python3.11/site-packages/slitlessutils/tests/test_acs_grism.py:114: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/home/runner/work/slitlessutils/slitlessutils/.tox/py311-test-cov/lib/python3.11/site-packages/slitlessutils/core/modules/module.py:114: in __call__
    results = pool(data, sources, total=len(data), **kwargs)
/home/runner/work/slitlessutils/slitlessutils/.tox/py311-test-cov/lib/python3.11/site-packages/slitlessutils/core/utilities/pool.py:143: in __call__
    with mp.Pool(processes=ncpu) as p:
/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/multiprocessing/context.py:119: in Pool
    return Pool(processes, initializer, initargs, maxtasksperchild,
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <multiprocessing.pool.Pool state=INIT pool_size=0>, processes = 0
initializer = None, initargs = (), maxtasksperchild = None
context = <multiprocessing.context.ForkContext object at 0x7f1d841e12d0>

    def __init__(self, processes=None, initializer=None, initargs=(),
                 maxtasksperchild=None, context=None):
        # Attributes initialized early to make sure that they exist in
        # __del__() if __init__() raises an exception
        self._pool = []
        self._state = INIT
    
        self._ctx = context or get_context()
        self._setup_queues()
        self._taskqueue = queue.SimpleQueue()
        # The _change_notifier queue exist to wake up self._handle_workers()
        # when the cache (self._cache) is empty or when there is a change in
        # the _state variable of the thread that runs _handle_workers.
        self._change_notifier = self._ctx.SimpleQueue()
        self._cache = _PoolCache(notifier=self._change_notifier)
        self._maxtasksperchild = maxtasksperchild
        self._initializer = initializer
        self._initargs = initargs
    
        if processes is None:
            processes = os.cpu_count() or 1
        if processes < 1:
>           raise ValueError("Number of processes must be at least 1")
E           ValueError: Number of processes must be at least 1

/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/multiprocessing/pool.py:205: ValueError
```

@Russell-Ryan , any insight into what's going on here? Are you generally running this on Mac or do you have an Ubuntu machine?

--- a/.github/workflows/ci_tests.yml
+++ b/.github/workflows/ci_tests.yml
@@ -31,12 +31,6 @@ jobs:
       matrix:
         include:
           - os: ubuntu-latest
-            python: '3.9'
-            tox_env: 'py39-test'
-            allow_failure: true
-            prefix: ''
-
-          - os: ubuntu-latest
             python: '3.10'
             tox_env: 'py310-test'
             allow_failure: false

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -31,7 +31,7 @@ classifiers = [
     'Topic :: Scientific/Engineering :: Astronomy',
 ]
 dynamic = ['version']
-requires-python = '>=3.9'
+requires-python = '>=3.10'
 dependencies = [
     'astropy>=5.0.4',
     'astroquery',

--- a/slitlessutils/core/utilities/pool.py
+++ b/slitlessutils/core/utilities/pool.py
@@ -126,6 +126,10 @@ class Pool:
         # get the number of CPUs to use
         ncpu = min(total, self.ncpu)
 
+        # Force to min 1
+        if ncpu == 0:
+            ncpu = 1
+
         # start multiprocessing as necessary
         if ncpu == 1:
             LOGGER.info('Serial processing')

